contrib/dhcp_agent/dhcp_agent.py

```diff
@@ -0,0 +1,350 @@
+        help='Name of the service to restart.',
+        default='isc-dhcp-server'
+    )
+    return parser
+
+
+def _setup_logging(filename, verbose=False):
+    log_size = 20  # MB
+    logger = logging.getLogger(__file__)
+    if verbose:
+        logger.setLevel(logging.INFO)
+    else:
+        logger.setLevel(logging.WARNING)
+    if not filename or filename in ('-', 'STDOUT'):
+        handler = logging.StreamHandler()
+    else:
+        handler = logging_handlers.RotatingFileHandler(
+            filename, maxBytes=(log_size * (1 << 20)), backupCount=5
+        )
+    fmt = logging.Formatter("[%(asctime)-12s.%(msecs)03d] "
+                            "%(levelname)-8s %(filename)s:%(lineno)d  "
+                            "%(message)s", "%Y-%m-%d %H:%M:%S")
+    handler.setFormatter(fmt)
+
+    logger.addHandler(handler)
+    return logger
+
+
+def _remove_apication_lock(lockfile, logger):
+    logger.info('Remove lock')
+    os.unlink(lockfile)
+
+
+def _set_script_lock(logger):
+    lockfile = '/tmp/{}.lock'.format(os.path.split(sys.argv[0])[1])
+    f = os.open(lockfile, os.O_TRUNC | os.O_CREAT | os.O_RDWR)
+    try:
+        fcntl.lockf(f, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        os.write(f, '{}'.format(os.getpid()))
+        atexit.register(_remove_apication_lock, lockfile, logger)
+    except IOError as e:
+        if e.errno == errno.EAGAIN:
+            logger.critical('Script already running.')
+            sys.exit(2)
+        raise
+
+
+def _check_and_return_params(parser):
+    required_params = {'host', 'key'}
+    params = parser.parse_args()[0]
+    diff = required_params - {k for k, v in params.__dict__.items() if v}
+    if diff:
+        parser.error('ERROR: {} are required.'.format(
+            ', '.join(['--{}'.format(d) for d in diff]))
+        )
+    if (
+        (params.dc and params.net_env) or
+        (not params.dc and not params.net_env)
+    ):
+        parser.error(
+            'ERROR: Only DC or ENV mode available.',
+        )
+    return params.__dict__
+
+
+class DHCPConfigManager(object):
+    def __init__(
+        self, logger, host, key, dc=None, net_env=None, mode=MODE_ALL,
+        verbose=False, restart=False, proto=PROTO_HTTPS,
+        dhcp_config_entries='', dhcp_config_networks='',
+        dhcp_service_name='isc-dhcp-server', **kwargs
+    ):
+        self.load_config()
+        self.logger = logger
+        self.host = host
+        self.key = key
+        self.proto = proto
+        self.restart = restart
+        self.dhcp_service_name = dhcp_service_name
+        self.dhcp_entries_config_path = dhcp_config_entries
+        self.dhcp_networks_config_path = dhcp_config_networks
+        self.envs = net_env.split(',') if net_env else []
+        self.dcs = dc.split(',') if dc else []
+        self.mode = mode
+
+    def run(self):
+        request_params = self.get_request_params(self.dcs, self.envs)
+        if self.mode == MODE_ALL:
+            for m in set(MODES) - set([MODE_ALL]):
```

Comment:
What is the meaning of `set(MODES) - set([MODE_ALL])`..? What about giving it a name?